ddtrace/contrib/internal/langchain/patch.py

```diff
@@ -965,6 +965,166 @@ def traced_similarity_search(langchain, pin, func, instance, args, kwargs):
     return documents
 
 
+@with_traced_module
+def traced_base_tool_invoke(langchain, pin, func, instance, args, kwargs):
+    integration = langchain._datadog_integration
+    tool_input = get_argument_value(args, kwargs, 0, "input")
+    config = get_argument_value(args, kwargs, 1, "config") if len(args) >= 2 else None
+
+    span = integration.trace(
+        pin,
+        "%s.%s.%s" % (func.__module__, func.__class__.__name__, func.__self__.__class__.__name__),
+        interface_type="tool",
+        submit_to_llmobs=True,
+    )
+
+    tool_output = None
+    try:
+        if instance.name:
+            span.set_tag_str("langchain.request.tool.name", integration.trunc(str(instance.name)))
+        if integration.is_pc_sampled_span(span):
+            if instance.description:
+                span.set_tag_str("langchain.request.tool.description", integration.trunc(str(instance.description)))
+            if instance.return_direct:
+                span.set_tag_str("langchain.request.tool.return_direct", integration.trunc(str(instance.return_direct)))
+            if instance.verbose:
+                span.set_tag_str("langchain.request.tool.verbose", integration.trunc(str(instance.verbose)))
+            if instance.metadata:
+                for key, value in instance.metadata.items():
+                    span.set_tag_str("langchain.request.tool.metadata.%s" % key, integration.trunc(str(value)))
+            if instance.tags:
+                for idx, tag in enumerate(instance.tags):
+                    span.set_tag_str("langchain.request.tool.tags.%d" % idx, integration.trunc(str(tag)))
+            if instance.handle_tool_error:
+                span.set_tag_str(
+                    "langchain.request.tool.handle_tool_error",
+                    integration.trunc(str(instance.handle_tool_error)),
+                )
+            if instance.handle_validation_error:
+                span.set_tag_str(
+                    "langchain.request.tool.handle_validation_error",
+                    integration.trunc(str(instance.handle_validation_error)),
+                )
+            if instance.response_format:
+                span.set_tag_str(
+                    "langchain.request.tool.response_format", integration.trunc(str(instance.response_format))
+                )
+            if tool_input:
+                span.set_tag_str("langchain.request.input", integration.trunc(str(tool_input)))
+            if config:
+                span.set_tag_str("langchain.request.config", integration.trunc(str(config)))
+        tool_output = func(*args, **kwargs)
+        if tool_output is not None:
+            span.set_tag_str("langchain.response.output", integration.trunc(str(tool_output)))
+    except Exception:
+        span.set_exc_info(*sys.exc_info())
+        integration.metric(span, "incr", "request.error", 1)
```

Comment:
No need for adding datadog metrics here, I think this is out of scope for this work and our team.